From dafe9a811fc78a53cca49df9921a10c9f6d730d4 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Mon, 30 Oct 2023 23:40:58 +0200
Subject: [PATCH] GLODROID: Report fake port if no one was detected

This allows devices without typec driver (rpi4, opi3) to be used in
gadget-only mode.

Change-Id: Ib30d70387c75c6a84c01950ae6441dddf2d863be
Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 usb/aidl/default/Usb.cpp | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/usb/aidl/default/Usb.cpp b/usb/aidl/default/Usb.cpp
index 593dd4f4c6..78a5970ad1 100644
--- a/usb/aidl/default/Usb.cpp
+++ b/usb/aidl/default/Usb.cpp
@@ -526,6 +526,25 @@ Status getPortStatusHelper(std::vector<PortStatus> *currentPortStatus) {
 
         return Status::SUCCESS;
     }
+
+    currentPortStatus->resize(1);
+    (*currentPortStatus)[0].portName = "port";
+    (*currentPortStatus)[0].canChangeMode = false;
+    (*currentPortStatus)[0].canChangeDataRole = false;
+    (*currentPortStatus)[0].canChangePowerRole = false;
+    (*currentPortStatus)[0].currentPowerRole = PortPowerRole::SINK;
+    (*currentPortStatus)[0].currentDataRole = PortDataRole::DEVICE;
+    (*currentPortStatus)[0].currentMode = PortMode::DRP;
+    (*currentPortStatus)[0].supportedModes.push_back(PortMode::DRP);
+    (*currentPortStatus)[0].usbDataStatus.push_back(UsbDataStatus::ENABLED);
+
+    ALOGI("Fake USB port: canChangeMode:%d canChagedata:%d canChangePower:%d "
+          "usbDataEnabled:%d",
+          (*currentPortStatus)[i].canChangeMode, (*currentPortStatus)[i].canChangeDataRole,
+          (*currentPortStatus)[i].canChangePowerRole, 0);
+
+    return Status::SUCCESS;
+
 done:
     return Status::ERROR;
 }
-- 
2.39.2

